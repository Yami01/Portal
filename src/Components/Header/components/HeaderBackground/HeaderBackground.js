/**
 *
 * HeaderBackground
 * @format
 * @flow
 *
 */

import React, { memo, useMemo } from 'react'
import type { Node, AbstractComponent } from 'react'
import Animated from 'react-native-reanimated'
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet'
import HeaderBackgroundStyles from './styles'

import type { PropsType } from './types'

import {
  DEFAULT_PADDING_VERTICAL,
  LARGE_PADDING_VERTICAL,
  STATUS_BAR_PADDING,
} from './config'
import { Colors } from '@/Theme/Variables'

const HeaderBackground: AbstractComponent<PropsType> = memo(
  (props: PropsType): Node => {
    const {
      children,
      testID,
      shouldIgnoreStatusBarHeight = false,
      extraStyle,
      extraPaddingBottom = 0,
      backgroundColor = Colors.red_950,
      withShadow = false,
      isLargePadding = false,
      onLayout,
    } = props

    const style = useMemo((): ViewStyleProp => {
      const paddingVertical: number = isLargePadding
        ? LARGE_PADDING_VERTICAL
        : DEFAULT_PADDING_VERTICAL
      const paddingBottom = Animated.add(paddingVertical, extraPaddingBottom)
      const paddingTop: number =
        paddingVertical + (shouldIgnoreStatusBarHeight ? 0 : STATUS_BAR_PADDING)
      return [
        HeaderBackgroundStyles.container,
        {
          paddingTop,
          paddingBottom,
          backgroundColor,
        },
        withShadow && HeaderBackgroundStyles.containerShadow,
        extraStyle,
      ]
    }, [
      backgroundColor,
      extraPaddingBottom,
      extraStyle,
      isLargePadding,
      shouldIgnoreStatusBarHeight,
      withShadow,
    ])

    return (
      <Animated.View style={style} testID={testID} onLayout={onLayout}>
        {children}
      </Animated.View>
    )
  },
)

export default HeaderBackground
