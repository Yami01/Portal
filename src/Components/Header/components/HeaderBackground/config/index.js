/**
 *
 * @format
 * @flow
 *
 */
// import WindowDimensions from '@core/Singleton/WindowDimensions'
// export const STATUS_BAR_PADDING: number = WindowDimensions.statusBarHeightInApp
export const STATUS_BAR_PADDING: number = 40

export const DEFAULT_PADDING_VERTICAL = 16
export const LARGE_PADDING_VERTICAL = 20
