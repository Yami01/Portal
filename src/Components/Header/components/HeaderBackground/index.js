/**
 *
 * @flow
 *
 */

export {default} from './HeaderBackground';
