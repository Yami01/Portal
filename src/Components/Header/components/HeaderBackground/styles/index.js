/**
 *
 * @format
 * @flow
 *
 */
import { StyleSheet } from 'react-native'

import type { StylePropsType } from '../types'
import { Colors } from '@/Theme/Variables'

const HeaderBackgroundStyles: StylePropsType = StyleSheet.create({
  container: {
    zIndex: 98,
    overflow: 'visible',
    borderBottomLeftRadius: 16,
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    backgroundColor: Colors.red_950,
    paddingHorizontal: 16,
  },
  containerShadow: {
    elevation: 12,
    shadowColor: Colors.dark_gray,
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.3,
    shadowRadius: 10,
  },
})

export default HeaderBackgroundStyles
