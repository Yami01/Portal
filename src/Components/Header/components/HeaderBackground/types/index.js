/**
 *
 * @format
 * @flow
 *
 */

import type { Node } from 'react'
import type { LayoutEvent } from 'react-native/Libraries/Types/CoreEventTypes'
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet'
import type { AnimatedValue } from 'react-native-reanimated'

export type PropsType = {
  children?: Node,
  extraStyle?: ViewStyleProp,
  isLargePadding?: boolean,
  shouldIgnoreStatusBarHeight?: boolean,
  extraPaddingBottom?: number | AnimatedValue<number>,
  backgroundColor?: string,
  withShadow?: boolean,
  testID?: string,
  onLayout?: (event: LayoutEvent) => any,
}

export type StylePropsType = {
  container: ViewStyleProp,
  containerShadow: ViewStyleProp,
}
