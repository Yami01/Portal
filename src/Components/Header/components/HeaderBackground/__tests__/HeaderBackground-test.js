/**
 *
 * HeaderBackground Test
 * @format
 * @flow
 *
 */

import React from 'react'
import { View } from 'react-native'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react-native'

import HeaderBackground from '../HeaderBackground'
import HeaderBackgroundStyles from '../styles'

const testID = 'headerBackgroundTestID'

describe('HeaderBackground', () => {
  test('render with default props', () => {
    const tree = renderer
      .create(
        <HeaderBackground>
          <View />
        </HeaderBackground>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render with shadow', () => {
    const tree = renderer
      .create(
        <HeaderBackground testID={testID} withShadow>
          <View />
        </HeaderBackground>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render with large padding', () => {
    const tree = renderer
      .create(
        <HeaderBackground isLargePadding>
          <View />
        </HeaderBackground>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render without status bar padding', () => {
    const tree = renderer
      .create(
        <HeaderBackground shouldIgnoreStatusBarHeight>
          <View />
        </HeaderBackground>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('is testID works', () => {
    const { getByTestId } = render(
      <HeaderBackground testID={testID}>
        <View />
      </HeaderBackground>,
    )
    expect(getByTestId(testID)).not.toBeNull()
  })

  test('is render properly without shadow', () => {
    const { getByTestId } = render(
      <HeaderBackground testID={testID}>
        <View />
      </HeaderBackground>,
    )
    const background = getByTestId(testID)
    expect(background).not.toBeNull()
    expect(background).not.toHaveStyle(HeaderBackgroundStyles.containerShadow)
  })

  test('is render properly with shadow', () => {
    const { getByTestId } = render(
      <HeaderBackground testID={testID} withShadow>
        <View />
      </HeaderBackground>,
    )
    const background = getByTestId(testID)
    expect(background).not.toBeNull()
    expect(background).toHaveStyle(HeaderBackgroundStyles.containerShadow)
  })
})
