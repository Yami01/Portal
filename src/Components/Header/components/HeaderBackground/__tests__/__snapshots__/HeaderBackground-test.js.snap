// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackground render with default props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`HeaderBackground render with large padding 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 20,
        },
        "paddingTop": 40,
      },
      false,
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`HeaderBackground render with shadow 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      Object {
        "elevation": 12,
        "shadowColor": "#363e3f",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
      },
      undefined,
    ]
  }
  testID="headerBackgroundTestID"
>
  <View />
</View>
`;

exports[`HeaderBackground render without status bar padding 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 16,
      },
      false,
      undefined,
    ]
  }
>
  <View />
</View>
`;
